fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/aspect/BulkEventAspect.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.event.business.aspect;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.apache.fineract.infrastructure.event.business.annotation.BulkEventSupport;
+import org.apache.fineract.infrastructure.event.business.service.BusinessEventNotifierService;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Aspect
+public class BulkEventAspect {
+
+    private final BusinessEventNotifierService businessEventNotifierService;
+
+    @Around("execution(* org.apache.fineract.commands.handler.NewCommandSourceHandler.processCommand(..)) && @within(bulkEventSupport)")
```

Comment:
I think we could remove the restriction to use this only on NewCommandSourceHandler interfaces, what if we just allow it everywhere?

Reply:
Modified using any class (with all methods) or only specific methods